### ISSUE TITLE
Modifying input_dim requires to update all transforms in the dataset_params.yaml file

### ISSUE THREAD

**Issue by utkutpcgl** [Deci-AI/super-gradients#986]
I just want to change in the input image size from the dataset_params of the dataset. For example for `from super_gradients.training.dataloaders.dataloaders import coco_detection_yolo_format_train`, I could not update the variables.

This is because the dataset_params.yaml file for coco_detection_yolo_format_base_dataset_params (here `super_gradients/recipes/dataset_params/coco_detection_yolo_format_base_dataset_params.yaml`) passes the harcoded input_dim to every transform in transforms as: `input_dim: ${dataset_params.train_dataset_params.input_dim}`. Dynamic updates in the code have no effect. 

This can cause silent errors if the programmer is not careful. This issue might be prevalent in other yaml files also.

**Comment by BloodAxe** [comment]
I agree that perhaps it's not that user friendly to override them like you wanted to. However the recipes were not intended for that use case. With existing configs one can reproduce the training and get the model's accuracy as advertised in the recipe. 

If you want to depart from this and train something on your own my best recommendation is to copy the configuration file and make necessary adjustments to it.

**Comment by utkutpcgl** [comment]
Thank you, that makes more sense. But the colab tutorial (https://colab.research.google.com/drive/1q0RmeVRzLwRXW-h9dPFSOchwJkThUy6d#scrollTo=jwy2Xq75dvfr) suggested that method. Currently, I am doing what you suggested.

**Comment by BloodAxe** [comment]
I totally hear your pan, and agree that modifying it in Colab is not user-friendly. 
![image](https://github.com/Deci-AI/super-gradients/assets/532320/981c58c1-33a0-460e-837a-e3c0f2609c3c)
